src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +50,29 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    -(-rational) == rational 
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    
+    (left+right) == (right+ left)
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    ((left - right) + right) == left
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    (left*right) == (right* left)
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
-    ???
+    left / right * right == left
   }
 
   property("division by zero") = forAll { (left: Rational, int: Int) =>
-    ???
+    true

```

Comment:
No idea what to test here...
throwing Exceptions?

Reply:
Exactly.
Review topics' tests to find the solution.